cabal-install/src/Distribution/Client/IndexUtils.hs

```diff
@@ -480,12 +462,32 @@ readRepoIndex verbosity repoCtxt repo idxState =
           return (mempty, mempty, emptyStateInfo)
         else ioError e
 
+    isOldThreshold :: Double
     isOldThreshold = 15 -- days
-    warnIfIndexIsOld dt = do
-      when (dt >= isOldThreshold) $ case repo of
-        RepoRemote{..} -> warn verbosity $ errOutdatedPackageList repoRemote dt
-        RepoSecure{..} -> warn verbosity $ errOutdatedPackageList repoRemote dt
-        RepoLocalNoIndex{} -> return ()
+    warnIfIndexIsOld isi = do
+      curTime <- getCurrentTime
+      case timestampToUTCTime (isiHeadTime isi) of
```

Comment:
This is a change in behaviour. Before it would warn the user if the index file has not been touched for 15 days (comparing `getCurrentTime` with `getModificationTime`). What you suggest warns the user when the index head time (latest known index state) is more than 15 days in the past.

I belive the old behaviour was more correct. Think of using a repo that only gets packages added once a month, you'd get cabal constantly telling you to do `cabal update` and doing `cabal update` won't even help, since you'd get the same `isiHeadTime`.

I don't think this change is relevant to the PR, so maybe you can just revert this bit? and back to
```
when (isRepoRemote repo) $ warnIfIndexIsOld =<< getIndexFileAge repo
```
?

Or is there some interaction I cannot see?